docs/metrics/gpt4-prompt-eng.md

```diff
@@ -0,0 +1,89 @@
+# GPT4 Prompt Engineering
+
+Prompting Engineering is a technique that involves building and optimizing specific input prompts to allow our Large Language Model (LLM) to perform better at targeted tasks. In the landscape of LLM evaluations, prompt engineering is designed to envoke the LLMs inherently knowledge and ability to reason to determine whether or not a set of inputs contain some sort of hallucination.
+
+## LLM Configurations
+
+Before diving into the techniques used for prompting we first need to understand some of the common terminology used as paramaters for LLMs.
+
+**Temperature** - Determines how the randomness of the model. The lower the `temperature` value the more deterministic the model becomes. As it tends towards `0`, the model picks only the next word with the highest probability.
+
+**Top k** - The model picks the next token from the list of top `k` tokens.
+
+**Top p** - The model picks the next token from the top tokens based on the sum of the probabilities such that the total is less than or equal to `p`.
+
+!!! warn "Limitations"
+
+    These metrics have limitations and should therefore be used at your own discretion.
+
+#  Prompting GPT4 for Hallucinations
+[GPT4](https://openai.com/gpt-4) is a closed LLM by OpenAI that we prompt in order to extract hallucinations.
+The initial specialized prompt passed into GPT4 includes the following:
+
+1. Asking GPT4 to determine in a binary fashion (strictly output yes or no) whether the generated answer contained hallucinations.
+2. Asking GPT4 for an explanation of why it answered either option.
+
+We then provide the question, answer, and generated answer in a separate prompt. We are also able to ask GPT4 to provide a score between `0.0` and `1.0` where `0.0`
+is a hallucination and `1.0` contains no hallucinated facts as well as a reason for why it gave a certain score.
+
+# Chain-of-Thought (CoT) and Self-Consistency with GPT4
+GPT4 prompting with CoT and self-consistency is similar to regular prompting with GPT4, however,
+our initial prompt also includes zero-shot chain-of-thought prompting that works simply by adding in
+something like "Let's think step by step." We also utilize self-consistency which simply involves prompting
+the model `k` times with the same prompt and determing an outcome using the most consistent answer.
+
+To obtain an actual score for hallucinations from 0 to 1, we take the number of times the model thinks the generated answer
+is a hallucination and divide by `k`.
+
+## Example
+Using OpenAI's API we would provide the following example system and user prompts to obtain some example output.
+
+Initial Prompt:
+```
+You will be provided with an open-domain problem.
```

Comment:
We try to keep each line to be < 120 char.  See other metrics doc as a reference.